packages/web-service/src/pages/habitat/check-habitat-answers/check-habitat-answers.js

```diff
@@ -1,6 +1,76 @@
 import pageRoute from '../../../routes/page-route.js'
 import { habitatURIs, TASKLIST } from '../../../uris.js'
+import { APIRequests } from '../../../services/api-requests.js'
+import { settTypes } from '../../../utils/sett-type.js'
+import { settDistruptionMethods } from '../../../utils/sett-disturb-methods.js'
+import Joi from 'joi'
+const addSett = 'additional-sett'
 
-export const completion = async _request => TASKLIST.uri
+const typeProcessor = selectedType => settTypes.filter(type => type.value === selectedType)[0].text
+const methodProcessor = selectedMethods => settDistruptionMethods.filter(method => selectedMethods.includes(method.value)).map(method => '\n' + method.text)
+export const dateProcessor = date => {
+  const dateObj = new Date(date)
+  const day = dateObj.getDate()
+  const year = dateObj.getFullYear()
+  return `${Number(day)} ${dateObj.toLocaleString('default', { month: 'long' })} ${year}`
+}
 
-export default pageRoute({ page: habitatURIs.CHECK_YOUR_ANSWERS.page, uri: habitatURIs.CHECK_YOUR_ANSWERS.uri, completion })
+export const checkData = async (request, h) => {
+  const journeyData = await request.cache().getData()
+
+  // Ensure if a user just deleted their only sett, we take them back to /tasklist
+  const habitatSites = await APIRequests.HABITAT.getHabitatsById(journeyData.habitatData.applicationId)
+
+  // Ensure the object is populated with the correct (and enough) keys
+  if (Object.keys(journeyData.habitatData).length !== 13 || habitatSites.length === 0) {
+    return h.redirect(TASKLIST.uri)
+  }
+  return undefined
+}
+
+export const getData = async request => {
```

Comment:
if I complete a site it adds it to the database as expected. I sign out and sign in and start another habitat journey. I am not taken directly to the check-your answers page but I am prompted to start adding